### PR TITLE
[WIP] Configure pod parallelism to higher value

### LINKED ISSUE #173: Configure default parallelism used by the workflow to a higher number to allow multiple executor pods to execute simultaneously
The default parallelism seems to be set to ~5 pods. This causes issues when dealing with 5+ pods.

Configure the parallelism using https://argoproj.github.io/argo-workflows/fields/#template field 

parallelism | integer | Parallelism limits the max total parallel pods that can execute at the same time in a workflow
-- | -- | --

### LINKED ISSUE #175: Increase the number of parallel workers processing HelmReleases in helm-operator
Increase the number of parallel workers processing HelmReleases in helm-operator so that a larger number of HelmRelease CROs can be reconciled simultaneously. The default is set to 4 workers.

### PR DESCRIPTION
**Pull request by nitishm**
Closes #173 
Closes #175 

Signed-off-by: Nitish Malhotra <nitish.malhotra@gmail.com>

### CODE REVIEW

**Comment by nitishm** [comment]
This needs to be verified externally with a more complex application group before this can be merged

File: pkg/workflow/argo.go [hunk 1]
```diff
@@ -33,7 +33,8 @@ const (
 )
 
 var (
-	timeout int64 = 3600
+	timeout            int64 = 3600
```
Comment (jonathan-innis) [inline pkg/workflow/argo.go:36]: We should convert these to `time.time` for readability

**Review by jonathan-innis — CHANGES_REQUESTED** [review]
Minor suggestion that has nothing to do with the point of the pr 😆

File: pkg/workflow/argo.go (hunk 1 above)
Comment (nitishm) [reply pkg/workflow/argo.go:36]: Will do

File: pkg/workflow/argo.go [hunk 2]
```diff
@@ -33,10 +34,18 @@ const (
 )
 
 var (
-	timeout            int64 = 3600
-	defaultParallelism int64 = 50
+	timeout            Timeout = "30m"
+	defaultParallelism int64   = 50
 )
 
+type Timeout string
```
Comment (nitishm) [inline pkg/workflow/argo.go:41]: @jonathan-innis Since I need a `*int64` I have to use a custom type. Can you think of another way of doing this?

**Review by davutkarabay — APPROVED** [review]

**Pull request merged**